src/test/java/com/glencoesoftware/omero/ms/image/region/ImageDataRequestHandlerTest.java

```diff
@@ -0,0 +1,690 @@
+import omeis.providers.re.Renderer;
+import omeis.providers.re.codomain.CodomainChain;
+import omeis.providers.re.codomain.CodomainMapContext;
+import omeis.providers.re.codomain.ReverseIntensityContext;
+import omero.ServerError;
+import omero.rtypes;
+import omero.api.RawPixelsStorePrx;
+import omero.model.Channel;
+import omero.model.ChannelI;
+import omero.model.DatasetI;
+import omero.model.DatasetImageLinkI;
+import omero.model.EventI;
+import omero.model.Experimenter;
+import omero.model.ExperimenterI;
+import omero.model.ImageI;
+import omero.model.LengthI;
+import omero.model.LogicalChannelI;
+import omero.model.ObjectiveI;
+import omero.model.ObjectiveSettingsI;
+import omero.model.Permissions;
+import omero.model.PermissionsI;
+import omero.model.PixelsI;
+import omero.model.ProjectDatasetLinkI;
+import omero.model.ProjectI;
+import omero.model.StatsInfoI;
+import omero.model.WellI;
+import omero.model.WellSampleI;
+import omero.model.enums.UnitsLength;
+
+import org.junit.Assert;
+import org.junit.Before;
+
+public class ImageDataRequestHandlerTest {
+
+    ImageI image;
+    PixelsI pixels;
+    EventI creationEvent;
+    Experimenter owner;
+    Optional<WellSampleI> wellSample;
+    Permissions permissions;
+    PixelBuffer pixelBuffer;
+    RawPixelsStorePrx rp;
+    Renderer renderer;
+    RenderingDef rdef;
+
+    public static String OWNER_FIRST_NAME = "firstname";
+    public static String OWNER_LAST_NAME = "lastname";
+    public static Long OWNER_ID = 123l;
+
+    public static long IMAGE_ID = 1l;
+    public static String IMAGE_NAME = "test name";
+    public static String IMAGE_DESC = "test image description";
+
+    public static long DATASET_LINK_ID_1 = 10l;
+
+    public static long DATASET_ID_1 = 100l;
+    public static String DATASET_NAME_1 = "ds name 1";
+    public static String DATASET_DESC_1 = "ds description 1";
+
+    public static long PROJECT_DS_LINK_ID_1 = 1000l;
+
+    public static long PROJECT_ID_1 = 10000l;
+    public static String PROJECT_NAME_1 = "proj name 1";
+    public static String PROJECT_DESC_1 = "proj description 1";
+
+    public static long PIXELS_ID = 12l;
+    public static String PIX_TYPE_STR = "uint8";
+
+    public static String CHANNEL_NAME_1 = "channel name 1";
+    public static String CHANNEL_NAME_2 = "channel name 2";
+    public static String CHANNEL_NAME_3 = "channel name 3";
+
+    public static Integer RES_LVL_COUNT = 5;
+    public static Integer TILE_WIDTH = 256;
+    public static Integer TILE_HEIGHT = 512;
+    public static int PIXELS_SIZE_X = 512;
+    public static int PIXELS_SIZE_Y = 1024;
+    public static int PIXELS_SIZE_Z = 1;
+    public static int PIXELS_SIZE_C = 3;
+    public static int PIXELS_SIZE_T = 1;
+
+    public static int BYTE_WIDTH = 1;
+
+    public static long WELL_SAMPLE_ID = 2l;
+    public static long WELL_ID = 20l;
+
+    public static double NOMINAL_MAGNIFICATION = 123.456;
+
+
+    JsonObject stdCorrect = new JsonObject("{" +
```

Comment:
This seems like it'll be a nightmare to maintain. Maybe put the expected JSON in a resource file and load it in?